actions/new_post.ts

```diff
@@ -68,57 +69,58 @@ export function completePostReceive(post: Post, websocketMessageProps: NewPostMe
                     now: Date.now(),
                 },
             },
-        ];
+            ...setChannelReadAndViewed(dispatch, getState, post, websocketMessageProps, fetchedChannelMember),
+        );
 
         dispatch(batchActions(actions));
 
-        // Still needed to update unreads
-
-        dispatch(setChannelReadAndViewed(post, websocketMessageProps, fetchedChannelMember));
-
         return dispatch(sendDesktopNotification(post, websocketMessageProps) as unknown as ActionFunc);
     };
 }
 
-export function setChannelReadAndViewed(post: Post, websocketMessageProps: NewPostMessageProps, fetchedChannelMember: boolean): ActionFunc {
-    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
-        const state = getState();
-        const currentUserId = getCurrentUserId(state);
-
-        // ignore system message posts, except when added to a team
-        if (shouldIgnorePost(post, currentUserId)) {
-            return {data: false};
+// setChannelReadAndViewed returns an array of actions to mark the channel read and viewed, and it dispatches an action
+// to asynchronously mark the channel as read on the server if necessary.
+export function setChannelReadAndViewed(dispatch: DispatchFunc, getState: GetStateFunc, post: Post, websocketMessageProps: NewPostMessageProps, fetchedChannelMember: boolean): Redux.AnyAction[] {
```

Comment:
While I'm explicitly avoiding extra dispatches here by passing around dispatch and getState, I don't want to get in a habit of doing that. I think it hurts readability, and I don't think it should make a performance difference in most places. The big problem seems to be if we get into a pattern like this where we make multiple changes to the same object in a row.

```javascript
const fooA = () => (dispatch) => {
    dispatch({type: 'CHANGE_A_THING'});
    dispatch(fooB());
};

const fooB = () => (dispatch) => {
    dispatch({type: 'CHANGE_A_THING'});
    dispatch(fooC());
};

const fooC = () => (dispatch) => {
    dispatch({type: 'CHANGE_A_THING'});
};
```

In this example, if fooA, fooB, and fooC all touched different objects which impact different parts of the UI, any performance hit would be much less noticeable.